cli/wallet/wallet.go

```diff
@@ -547,22 +547,49 @@ func importMultisig(ctx *cli.Context) error {
 		l := ctx.String("name")
 		label = &l
 	}
-	acc, err := newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
-	if err != nil {
-		return cli.NewExitError(err, 1)
-	}
 
-	if err := acc.ConvertMultisig(m, pubs); err != nil {
-		return cli.NewExitError(err, 1)
-	}
+	var acc *wallet.Account
 
+	// Check if a WIF is provided, and if so, create an account from it
+	if ctx.IsSet("wif") {
+		acc, err = newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+	} else {
+		// Otherwise, check if any of the provided public keys correspond to an existing key in the wallet
+		for _, pub := range pubs {
+			existAccount, err := findAccountByPublicKey(wall, pub)
+			if err == nil {
+				acc = existAccount
+				break
+			}
+		}
```

Comment:
We need to ensure that matching account is the only one. If there are more accounts, then we need to fallback to WIF.

Reply:
Place this check firstly. After that if you still need WIF, then add `if ctx.IsSet("wif")` check or error if it's not provided.